src/tools/website-components/Playground.jsx

```diff
@@ -1,100 +1,167 @@
+    const playgroundUrl =
+      playgroundLink ||
+      encodeURI(
+        `/storybook/${system}/index.html?selectedKind=${selectedKind}&selectedStory=${selectedStory}&stories=1`
+      );
+
+    const fullFrameUrl =
+      system && selectedKind && selectedStory
+        ? encodeURI(
+            `/storybook/${system}/iframe.html?selectedKind=${selectedKind}&selectedStory=${selectedStory}`
+          )
+        : '';
+
+    return (
+      <div className={styles.playground} ref={this.playgroundRef}>
+        <div className={styles.showcase}>
+          {!hideDemo && (
+            <Fragment>
+              {showFrame && fullFrameUrl ? (
+                <Iframe url={fullFrameUrl} defaultHeight={frameHeight} />
+              ) : (
+                <div className={styles.showcase__content}>{children}</div>
+              )}
+            </Fragment>
           )}
-        </Fragment>
-      )}
-      <ul
-        className={classnames(styles.links, {
-          [styles['links--showcase-hidden']]: hideDemo,
-        })}
-      >
-        {playgroundUrl && (
-          <li className={styles['link-item']}>
-            <a
-              href={playgroundUrl}
-              className={styles.link}
-              target="_blank"
-              rel="noopener noreferrer"
-            >
-              Playground
-            </a>
-          </li>
-        )}
-        {fullFrameUrl && (
-          <li className={styles['link-item']}>
+
+          {fullFrameUrl && (
             <a
               href={fullFrameUrl}
-              className={styles.link}
+              className={`${styles.link} ${styles['link--icon']} ${
+                styles.fullscreen
+              }`}
               target="_blank"
               rel="noopener noreferrer"
             >
-              Fullscreen
+              <span className={styles.link__label}>Fullscreen</span>
+              <svg className={styles.link__icon}>
+                <use xlinkHref={`${iconSprite}#ui--fullscreen`} />
+              </svg>
             </a>
-          </li>
+          )}
+        </div>
+
+        <div className={styles.code}>
+          <pre
+            className={`${styles['code-pre']} language-html`}
+            ref={this.showcaseCodeRef}
+          >
+            <code
+              className="language-html"
+              dangerouslySetInnerHTML={{
+                __html: Prism.highlight(
+                  beautifyHtml(ReactDOMServer.renderToStaticMarkup(children), {
+                    indent_size: 2,
+                    wrap_line_length: 120,
+                  }),
+                  Prism.languages.html,
+                  'html'
+                ),
+              }}
+            />
+          </pre>
+
+          <button
+            type="button"
+            className={`${styles.link} ${styles['link--icon']} ${
+              styles.toggle
+            }`}
+            onClick={this.handleClickOnToggle.bind(this)}
```

Comment:
Bind in the constructor, not in the render ;)

`this.handleClickOnToggle = this.handleClickOnToggle.bind(this);`